buildSrc/src/main/kotlin/CodegenTestCommon.kt

```diff
@@ -187,19 +189,23 @@ fun Project.registerModifyMtimeTask() {
         dependsOn("generateSmithyBuild")
 
         doFirst {
-            @Suppress("UNCHECKED_CAST") val previousBuildHashes: Map<String, Long> = project.extra["previousBuildHashes"] as Map<String, Long>
-
-            project.buildDir.walk()
-                .filter { it.isFile }
-                .map {
-                    getChecksumForFile(it) to it
-                }
-                .forEach { (currentHash, currentFile) ->
-                    previousBuildHashes[currentHash]?.also { oldMtime ->
-                        println("Setting `mtime` of $currentFile back to `$oldMtime` because its hash `$currentHash` remained unchanged after a rebuild.")
-                        currentFile.setLastModified(oldMtime)
+            if (!project.extra.has(previousBuildHashesKey)) {
+                println("No hashes from a previous build exist because `generateSmithyBuild` is up to date, skipping `mtime` fixups")
```

Comment:
side note—I'm actually not sure the best thing to do here, but we might want to use some sort of logger invocation instead of println

Reply:
In Gradle, you should use SLF4J to log: https://docs.gradle.org/current/userguide/logging.html#sec:sending_your_own_log_messages